airflow/metrics/otel_logger.py

```diff
@@ -0,0 +1,234 @@
+            return counter.add(value, attributes=tags)
+
+    @validate_stat
+    def decr(self, stat: str, count: int = 1, rate: float = 1, tags: Attributes = None):
+        """
+        Decrement stat by count.
+
+        :param stat: The name of the stat to increment.
+        :param count: A positive integer to subtract from current value of stat.
+        :param rate: TODO: define me
+        :param tags: Tags to append to the stat.
+        """
+        if (count < 0) or (rate < 0):
+            raise ValueError("count and rate must both be positive values.")
+        # TODO: I don't think this is the right use for rate???
+        value = count * rate
+
+        if self.metrics_validator.test(stat):
+            counter = self.metrics_map.get_counter(f"{self.prefix}.{stat}")
+            return counter.add(-value, attributes=tags)
+
+    @validate_stat
+    def gauge(
+        self,
+        stat: str,
+        value: int | float,
+        rate: float = 1,
+        delta: bool = False,
+        *,
+        tags: Attributes = None,
+    ) -> None:
+        """Gauge stat."""
+        # To be implemented
+        return None
+
+    @validate_stat
+    def timing(
+        self,
+        stat: str,
+        dt: DeltaType,
+        *,
+        tags: Attributes = None,
+    ) -> None:
+        """Stats timing."""
+        # To be implemented
+        return None
+
+    @validate_stat
+    def timer(
+        self,
+        stat: str | None = None,
+        *args,
+        tags: Attributes = None,
+        **kwargs,
+    ) -> TimerProtocol:
+        """Timer metric that can be cancelled."""
+        # To be implemented
+        return Timer()
+
+
+class BaseInstrument(Instrument):
+    """Instrument clss is abstract and must be implemented."""
+
+    def __init__(self, name: str, unit: str = "", description: str = "", attributes: Attributes = None):
+        self.name: str = name
+        self.unit: str = unit
+        self.description: str = description
+        self.attributes: Attributes = attributes
+
+
+class MetricsMap:
+    """Stores Otel Counters."""
+
+    def __init__(self, meter):
+        self.meter = meter
+        self.map = {}
+
+    def clear(self) -> None:
+        self.map.clear()
+
+    def _create_counter(self, name):
+        """Creates a new counter or up_down_counter for the provided name."""
+        otel_safe_name = name[:OTEL_NAME_MAX_LENGTH]
+        if name != otel_safe_name:
+            warnings.warn(
+                f"Metric name `{name}` exceeds OpenTelemetry's name length limit of "
+                f"{OTEL_NAME_MAX_LENGTH} characters and will be truncated to `{otel_safe_name}`."
+            )
+
+        if _is_up_down_counter(name):
+            counter = self.meter.create_up_down_counter(name=otel_safe_name)
+        else:
+            counter = self.meter.create_counter(name=otel_safe_name)
+
+        print(f"--> created {otel_safe_name} as type: {str(type(counter)).split('.')[-1][:-2]}")
```

Comment:
This was for my own debugging use, I can delete it when I make other revisions.

Reply:
You should be able to instantiate a logger in this module, that'd be preferable over using `print`

Reply:
Yeah, I think I'll leave it in as a logging.info() unless someone objects.

Reply:
Made them log messages in https://github.com/apache/airflow/pull/30873/commits/683b2bc77a4d6ff5fc9d9dd6a8c300a973582616